source/mge.js

```diff
@@ -0,0 +1,988 @@
+                let content = '';
+                if (this.const_val !== null) {content += this.const_val;}
+                else {content += '[';}
+                for (var t of Object.values(this.type)) {
+                    content += t.__name__;
+                }
+                content += ']';
+                return content;
+            }
+
+            formatString() {
+                if (this.const_val !== null) {return this.const_val;}
+                else
+                if (this.type[0].__module__ !== undefined) {return this.type[0].__name__;}
+                return "None";
+            }
+        });
+        this.registerType('megengine.tensor.Tensor', class {
+            constructor(data, dtype, device) {
+                this.data = data;
+                this.dtype = dtype;
+                this.device = device;
+            }
+        });
+
+        this.registerType('megengine.core.tensor.dtype.QuantDtypeMeta', class {
+            constructor(name, cname, np_dtype, qmin, qmax, is_signed) {
+                this.name = name;
+                this.cname = cname;
+                this.np_dtype = np_dtype;
+                this.qmin = qmin;
+                this.qmax = qmax;
+                this.is_signed = is_signed;
+            }
+        });
+
+    }
+};
+
+megengine.Container = class {
+
+    static open(context) {
+
+        const pickle = megengine.Container.Pickle.open(context.stream);
+        if (pickle) {
+            return pickle;
+        }
+
+        return null;
+    }
+};
+
+
+megengine.Container.Pickle = class {
+
+    static open(stream) {
+        return new megengine.Container.Pickle(stream);
+    }
+
+    constructor(stream) {
+        this._stream = stream;
+        this._graphs = [this];
+    }
+
+    set metadata(value) {
+        this._metadata = value;
+    }
+
+    set exception(value) {
+        this._exceptionCallback = value;
+    }
+
+    get format() {
+        this._unpickle();
+        var version = "1.6";
+        var dump_info = this._graphs[0].data.dump_info;
+        if (dump_info !== undefined) {
+            version = dump_info.version;
+        }
+        return 'megengine v' + version;
+    }
+
+    get graphs() {
+        this._unpickle();
+        return this._graphs;
+    }
+
+    get littleEndian() {
+        this._unpickle();
+        return this._littleEndian;
+    }
+
+    _unpickle() {
+        if (!this._stream) {
+            return;
+        }
+
+        const data = this._stream.length < 0x7ffff000 ? this._stream.peek() : this._stream;
+        const execution = new megengine.Execution(null, this._exceptionCallback);
+        const unpickler = python.Unpickler.open(data, execution);
+
+        this._stream = null;
+        this._exceptionCallback = null;
+
+        const module_source_map = new Map();
+        const deserialized_objects = new Map();
+        unpickler.persistent_load = (saved_id) => {
+            const typename = saved_id.shift();
```

Comment:
Copied code that is not being used? Please review and remove unused code.

Reply:
done